yarn-project/acir-simulator/src/client/client_execution.ts

```diff
@@ -0,0 +1,57 @@
+import { ACVMField, toACVMField, fromACVMField, toAcvmNoteLoadOracleInputs, createDummyNote } from '../acvm/index.js';
+import { AztecAddress, Fr } from '@aztec/foundation';
+import {
+  CallContext,
+  PrivateHistoricTreeRoots,
+  TxRequest,
+  FunctionData,
+  PRIVATE_DATA_TREE_HEIGHT,
+} from '@aztec/circuits.js';
+import { DBOracle } from './db_oracle.js';
+import { FunctionAbi } from '@aztec/noir-contracts';
+
+export abstract class ClientExecution<T> {
+  constructor(
```

Comment:
nit: I prefer to not use base classes if we can help it. It just makes skimming the class not enough, then have to go one layer to see all it can do. Won't push it, but I'd prefer if this was just a regular class that is used by the other classes.